server/server.go

```diff
@@ -1,35 +1,43 @@
 package server
 
 import (
+	"errors"
 	"fmt"
 	"net/http"
-	"time"
+	"strings"
 )
 
-// Middleware is a function that decorates an http.Handler
-type Middleware func(handler http.Handler) http.Handler
+var (
+	// SupportedAuthTypes is a whitelist of supported authentication types. Deafult: Bearer
+	SupportedAuthTypes = [...]string{"Bearer"}
+	// ErrorInvalidAuthorizationHeader is a static error for invalid authorization
+	ErrorInvalidAuthorizationHeader = errors.New("InvalidAuthorizationHeader")
+	// ErrorUnsupportedAuthorizationType is a static error returned if the auth type is not in the whitelist
+	ErrorUnsupportedAuthorizationType = fmt.Errorf("UnsupportedAuthorizationType, supported types are %v", SupportedAuthTypes)
+	// ErrorInvalidAuthToken is a static error returned when authorization fails
+	ErrorInvalidAuthToken = errors.New("InvalidAuthToken")
+)
 
-// ApplyMiddleware creates a final http Handler with a slice of middleware functions applied
-func ApplyMiddleware(handlers []Middleware, final http.Handler) http.Handler {
-	for _, handler := range handlers {
-		final = handler(final)
+// ExtractBearerToken attempts to extract an Oauth bearer token
+// from the provided request, returning extracted token and error (if any)
+func ExtractBearerToken(r *http.Request) (string, error) {
```

Comment:
Please cleanup the now deprecated version of this in https://github.com/tozny/utils-go/blob/master/http.go